en/appendices/5-0-migration-guide.rst

```diff
@@ -87,6 +71,10 @@ Database
 - ``Driver::quote()`` was removed. Use prepared statements instead.
 - ``Query::orderBy()`` was added to replace ``Query::order()``.
 - ``Query::groupBy()`` was added to replace ``Query::group()``.
+- The ``DateTimeType`` and ``DateType`` now always return immutable objects.
+  Additionally the interface for ``Date`` objects reflects the ``ChronosDate``
+  interface which lacks all of the time related methods that were present in
+  CakePHP 4.x.
```

Comment:
This is pretty brief, as my hope is that we front-load most of these changes as deprecations into 4.5. We might not be able to make the backport fully typesound, but we will be able to emit deprecations on methods that will be removed and ensure that the mutable variants also emit deprecations.

My plan is to have the 4.5 version of date & datetimes quack enough like what is going to be present in 5.x that the breaking changes are minimal if they exist at all.